### PR TITLE
Add a rule to check for ICMP ingress traffic allowed to all

### PR DESCRIPTION
**Pull request by gebailey**
For AWS provider, provide a rule that checks for ICMP traffic being allowed from all locations.  By default this rule is disabled in both the default and detailed rulesets.

### CODE REVIEW

File: ScoutSuite/providers/base/authentication_strategy_factory.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 }
 
 
+class InvalidAuthenticationStrategyException(Exception): pass
```
Comment (j4v) [inline ScoutSuite/providers/base/authentication_strategy_factory.py]: Why this?

**Comment by Aboisier** [comment]
Hey Greg, I moved your PR target to develop (the contribution process has been formalized a few months ago and is described in the [CONTRIBUTING.md](https://github.com/nccgroup/ScoutSuite/blob/master/CONTRIBUTING.md)). 

Thanks for your contribution! I tested it, looks good.

File: ScoutSuite/providers/base/authentication_strategy_factory.py (hunk 1 above)
Comment (Aboisier) [reply ScoutSuite/providers/base/authentication_strategy_factory.py]: Don't worry, it's just a bad merge, I'm fixing this.

**Comment by gebailey** [comment]
> Hey Greg, I moved your PR target to develop (the contribution process has been formalized a few months ago and is described in the [CONTRIBUTING.md](https://github.com/nccgroup/ScoutSuite/blob/master/CONTRIBUTING.md)).
> 
> Thanks for your contribution! I tested it, looks good.

Thanks for the pointer to CONTRIBUTING.md; obviously should have read that first!  :-)

I can't figure out why the travis CI build failed, though... is that a transient issue?  It's not clear how adding a rule would cause other parts to fail?

**Comment by Aboisier** [comment]
> Thanks for the pointer to CONTRIBUTING.md; obviously should have read that first! :-)

No worries!

> I can't figure out why the travis CI build failed, though... is that a transient issue? It's not clear how adding a rule would cause other parts to fail?

We are indeed having a flaky build situation. I restarted the build and it passed. It's been happening for a month or so, not sure what is causing this.

**Comment by j4v** [comment]
Reviewed and approved, thanks @gebailey :medal_sports:

**Pull request merged**